python/link_prediction.py

```diff
@@ -0,0 +1,970 @@
+        hidden_features_size: mgp.List[int] -> Defines the size of each hidden layer in the architecture.
+        layer_type: str -> Layer type
+        num_epochs: int -> Number of epochs for model training
+        optimizer: str -> Can be one of the following: ADAM, SGD, AdaGrad...
+        learning_rate: float -> Learning rate for optimizer
+        split_ratio: float -> Split ratio between training and validation set. There is not test dataset because it is assumed that user first needs to create new edges in dataset to test a model on them.
+        node_features_property: str → Property name where the node features are saved.
+        device_type: str ->  If model will be trained using CPU or cuda GPU. Possible values are cpu and cuda. To run it on Cuda, user must set this flag to true and system must support cuda execution.
+                                System's support is checked with torch.cuda.is_available()
+        console_log_freq: int ->  how often do you want to print results from your model? Results will be from validation dataset.
+        checkpoint_freq: int → Select the number of epochs on which the model will be saved. The model is persisted on disc.
+        aggregator: str → Aggregator used in models. Can be one of the following: lstm, pool, mean, gcn.
+        metrics: mgp.List[str] -> Metrics used to evaluate model in training.
+        predictor_type str: Type of the predictor. Predictor is used for combining node scores to edge scores.
+        attn_num_heads: List[int] -> GAT can support usage of more than one head in each layer except last one. Only used in GAT, not in GraphSage.
+        tr_acc_patience: int -> Training patience, for how many epoch will accuracy drop on test set be tolerated before stopping the training.
+        context_save_dir: str -> Path where the model and predictor will be saved every checkpoint_freq epochs.
+        target_relation: str -> Unique edge type that is used for training.
+        num_neg_per_pos_edge (int): Number of negative edges that will be sampled per one positive edge in the mini-batch.
+        num_layers (int): Number of layers in the GNN architecture.
+        batch_size (int): Batch size used in both training and validation procedure.
+        sampling_workers (int): Number of workers that will cooperate in the sampling procedure in the training and validation.
+
+    Returns:
+        mgp.Record:
+            status (bool): True if parameters were successfully updated, False otherwise.
+            message(str): Additional explanation why method failed or OK otherwise.
+    """
+    global link_prediction_parameters
+
+    _validate_user_parameters(parameters=parameters)
+    
+    for key, value in parameters.items():
+        if not hasattr(link_prediction_parameters, key):
+            return mgp.Record(
+                status=1,
+                message="No attribute " + key + " in class LinkPredictionParameters",
```

Comment:
I think we shouldn't mention LinkPredictionParameters class in the error message